### PR TITLE
Permissions emulation

### PR DESCRIPTION
**Pull request by justnero**

### CODE REVIEW

File: openapi/components/schemas/Profile.yaml [hunk 1]
```diff
@@ -71,3 +71,6 @@ properties:
       User preferences like timezone, language and many more. This is an object
       with custom properties.
     type: object
+  isPermissionsEmulation:
```
Comment (adamaltman) [inline openapi/components/schemas/Profile.yaml:74]:
```suggestion
  hasPermissionsEmulation:
```
I'm looking for something that is grammatically correct?

File: openapi/components/schemas/Profile.yaml [hunk 2]
```diff
@@ -71,3 +71,6 @@ properties:
...
+    type: boolean
+    description: Is the current user session has permissions emulation enabled.
```
Comment (adamaltman) [inline openapi/components/schemas/Profile.yaml:76]:
```suggestion
    description: True when the current user session has permissions emulation enabled.
```

File: openapi/paths/permissions-emulation.yaml [hunk 3]
```diff
@@ -0,0 +1,75 @@
+post:
+  tags:
+    - Profile
+  summary: Start permissions emulation
+  operationId: PostPermissionsEmulation
+  description: |
+    Start permissions emulation.
+
+    Emulation during emulation is not supported. If request sent during an ongoing emulation 403 sent in response.
```
Comment (adamaltman) [inline openapi/paths/permissions-emulation.yaml:9]:
```suggestion
    Emulation during emulation is not supported. If request sent during an ongoing emulation 403 sent in response.
    
    Escalation is also not supported.
```

Am I correct that escalation is not supported?

Comment (justnero) [reply openapi/paths/permissions-emulation.yaml:9]: Yes, sure.

**Review by ashkarpetin — DISMISSED** [review]

File: openapi/paths/permissions-emulation.yaml [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+  summary: Stop permissions emulation
+  operationId: DeletePermissionsEmulation
+  description: |
+    Stop permissions emulation.
+
+    Session permissions would be restored to the state before emulation started.
+    If emulations was not started `403` is sent in response.
+  responses:
+    '201':
+      description: Session was created.
```
Comment (iamdevice) [inline openapi/paths/permissions-emulation.yaml:62]:
```suggestion
      description: Session was restored.
```

**Review by iamdevice — APPROVED** [review]

**Review by ashkarpetin — APPROVED** [review]

**Pull request merged**